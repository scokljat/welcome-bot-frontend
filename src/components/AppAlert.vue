<template>
  <div class="alert" :class="{ success: alert.success, visible: alert.active }">
    <i :class="alertIcon"></i>
    <span>{{ alert.message }}</span>
  </div>
</template>

<script>
import { mapActions, mapGetters } from 'vuex';

export default {
  name: 'AppAlert',
  computed: {
    ...mapGetters({ alert: 'getAlert' }),
    alertIcon() {
      return this.alert.success
        ? 'fa-solid fa-circle-check'
        : 'fa-solid fa-circle-exclamation';
    },
  },
  mounted() {
    setTimeout(() => this.hideAlert({ active: false }), 3000);
  },
  methods: {
    ...mapActions({ hideAlert: 'hideAlert' }),
  },
};
</script>

<style lang="scss">
.alert {
  position: fixed;
  display: flex;
  background-color: #d9534f;
  justify-content: center;
  align-items: center;
  top: 1.5rem;
  left: 50%;
  transform: translateX(-50%);
  padding: 1.2rem 1rem;
  color: white;
  z-index: 1000;
  border-radius: 0.25rem;
  min-width: 70%;
  opacity: 0;
  visibility: hidden;
  transition: opacity 1.2s ease-in-out;

  @include tablet-md {
    min-width: unset;
  }

  i {
    margin-right: 1rem;
  }

  span {
    width: max-content;
  }
}

.alert.success {
  background-color: #04aa6d;
}

.visible {
  visibility: visible;
  opacity: 1;
}
</style>
