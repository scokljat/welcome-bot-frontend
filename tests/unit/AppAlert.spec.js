import { render } from '@testing-library/vue';
import '@testing-library/jest-dom';
import AppAlert from '@/components/AppAlert';
import { createStore } from 'vuex';

const store = createStore({
  state() {
    return {
      alert: {
        active: true,
        success: false,
        message: 'Error message',
      },
    };
  },
  getters: {
    getAlert(state) {
      return state.alert;
    },
  },
});

store.commit = jest.fn();

const factory = (props) =>
  render(AppAlert, {
    props,
    global: {
      plugins: [store],
    },
  });

describe('AppAlert.vue', () => {
  test('renders properly', () => {
    const wrapper = factory();

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders error message', () => {
    const wrapper = factory();

    expect(wrapper.getByText('Error message'));
  });

  test('renders errors alert icon', () => {
    const wrapper = factory();

    expect(wrapper.getByTestId('icon')).toHaveClass(
      'fa-solid fa-circle-exclamation'
    );
  });

  test('renders success alert icon', () => {
    store.state.alert.success = true;
    const wrapper = factory();

    expect(wrapper.getByTestId('icon')).toHaveClass('fa-solid fa-circle-check');
  });
});
