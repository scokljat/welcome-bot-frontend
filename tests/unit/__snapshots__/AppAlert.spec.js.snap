// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppAlert.vue renders properly 1`] = `<div class="alert visible"><i data-testid="icon" class="fa-solid fa-circle-exclamation"></i><span>Error message</span></div>`;
